src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1228,70 +1289,66 @@ private async ValueTask<Http2Stream> SendHeadersAsync(HttpRequestMessage request
+                            // and perform another connection attempt. The user should never see this exception.
+                            s.thisRef.ThrowShutdownException();
+                        }
 
-                            // Now that we're holding the lock, configure the stream.  The lock must be held while
-                            // assigning the stream ID to ensure only one stream gets an ID, and it must be held
-                            // across setting the initial window size (available credit) and storing the stream into
-                            // collection such that window size updates are able to atomically affect all known streams.
-                            s.http2Stream.Initialize(s.thisRef._nextStream, _initialWindowSize);
+                        // Now that we're holding the lock, configure the stream.  The lock must be held while
+                        // assigning the stream ID to ensure only one stream gets an ID, and it must be held
+                        // across setting the initial window size (available credit) and storing the stream into
+                        // collection such that window size updates are able to atomically affect all known streams.
+                        s.http2Stream.Initialize(s.thisRef._nextStream, _initialWindowSize);
 
-                            // Client-initiated streams are always odd-numbered, so increase by 2.
-                            s.thisRef._nextStream += 2;
+                        // Client-initiated streams are always odd-numbered, so increase by 2.
+                        s.thisRef._nextStream += 2;
 
-                            // We're about to flush the HEADERS frame, so add the stream to the dictionary now.
-                            // The lifetime of the stream is now controlled by the stream itself and the connection.
-                            // This can fail if the connection is shutting down, in which case we will cancel sending this frame.
-                            s.thisRef._httpStreams.Add(s.http2Stream.StreamId, s.http2Stream);
-                        }
+                        // We're about to flush the HEADERS frame, so add the stream to the dictionary now.
+                        // The lifetime of the stream is now controlled by the stream itself and the connection.
+                        // This can fail if the connection is shutting down, in which case we will cancel sending this frame.
+                        s.thisRef._httpStreams.Add(s.http2Stream.StreamId, s.http2Stream);
+                    }
 
-                        Span<byte> span = writeBuffer.Span;
+                    Span<byte> span = writeBuffer.Span;
+
+                    // Copy the HEADERS frame.
+                    ReadOnlyMemory<byte> current, remaining;
+                    (current, remaining) = SplitBuffer(headerBytes, FrameHeader.MaxPayloadLength);
```

Comment:
```suggestion
                    (current, remaining) = SplitBuffer(s.headerBytes, FrameHeader.MaxPayloadLength);
```

Reply:
(It'd be good to look at the IL for this and ensure there's no accidental closure here.  I'm hoping in the very near future the compiler will be able to check that for us via static lambdas.)

Reply:
Seems like the safe thing to do would be to split it into a separate function.

On the other hand, if compiler support is imminent, then it's wasted work...

Reply:
@RikkiGibson, I see https://github.com/dotnet/roslyn/pull/45835 was just merged.  Does that mean that if dotnet/runtime ingests an updated SDK / Roslyn build, we'll be able to use static lambdas?

Reply:
Yes, the static lambdas feature should be available once you ingest the updated build.

Reply:
Thanks, @RikkiGibson.